driver-core/src/test/functional/com/mongodb/client/model/AggregatesTest.java

```diff
@@ -0,0 +1,178 @@
+import com.mongodb.client.model.geojson.Point;
+import com.mongodb.client.model.geojson.Position;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.bson.Document;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.bson.codecs.DecoderContext;
+import org.bson.conversions.Bson;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+import static com.mongodb.client.model.GeoNearOption.distanceMultiplier;
+import static com.mongodb.client.model.GeoNearOption.includeLocs;
+import static com.mongodb.client.model.GeoNearOption.key;
+import static com.mongodb.client.model.GeoNearOption.maxDistance;
+import static com.mongodb.client.model.GeoNearOption.minDistance;
+import static com.mongodb.client.model.GeoNearOption.query;
+import static com.mongodb.client.model.GeoNearOption.spherical;
+import static com.mongodb.client.model.Aggregates.geoNear;
+import static com.mongodb.client.model.Aggregates.unset;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+public class AggregatesTest extends OperationFunctionalSpecification {
+
+    @BeforeEach
+    public void beforeEach() {
+        super.setupInternal();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        super.cleanupInternal();
+    }
+
+    void insertAll(final String insertAll) {
+        List<BsonDocument> documents = BsonArray.parse(insertAll).stream().map(BsonValue::asDocument).collect(Collectors.toList());
+        getCollectionHelper().insertDocuments(documents);
+    }
+
+    private List<Bson> assertPipeline(final String stageAsString, final Bson stage) {
+        BsonDocument expectedStage = BsonDocument.parse(stageAsString);
+        List<Bson> pipeline = Collections.singletonList(stage);
+        assertEquals(expectedStage, pipeline.get(0).toBsonDocument(BsonDocument.class, getDefaultCodecRegistry()));
+        return pipeline;
+    }
+
+    private void assertResults(final List<Bson> pipeline, final String s) {
+        List<Document> expectedResults = parseToList(s);
+        List<Document> results = getCollectionHelper().aggregate(pipeline);
+        assertEquals(expectedResults, results);
+    }
+
+    private List<Document> parseToList(final String s) {
+        return BsonArray.parse(s).stream().map(v -> toDocument(v.asDocument())).collect(Collectors.toList());
+    }
+
+    public Document toDocument(final BsonDocument bsonDocument) {
+        return getDefaultCodecRegistry().get(Document.class).decode(bsonDocument.asBsonReader(), DecoderContext.builder().build());
+    }
+
+    @Test
+    public void testUnset() {
+        assumeTrue(serverVersionAtLeast(4, 2));
+        insertAll("[\n"
```

Comment:
Keeping the `\n` allows opt-enter > copy concat to clipboard (or edit in separate window) while preserving linebreaks. Otherwise everything ends up on one line if someone needs to edit or test something.